src/zope/publisher/browser.py

```diff
@@ -277,69 +277,83 @@ def _decode(self, text):
 
     def processInputs(self):
         'See IPublisherRequest'
+        items = []
+
+        # We could simply not parse QUERY_STRING if it's absent, but this
+        # provides slightly better doctest-compatibility with the old code
+        # based on cgi.FieldStorage.
+        self._environ.setdefault('QUERY_STRING', '')
+
+        if self.method in _get_or_head:
+            # PEP-3333 specifies that strings must only contain codepoints
+            # representable in ISO-8859-1.
+            kwargs = {}
+            if not PYTHON2:
+                kwargs['encoding'] = 'ISO-8859-1'
+                kwargs['errors'] = 'replace'
+            items.extend(parse_qsl(
+                self._environ['QUERY_STRING'], keep_blank_values=True,
+                **kwargs))
+        elif self.method not in _get_or_head:
+            env = self._environ.copy()
+            env['wsgi.input'] = self._body_instream
+            # cgi.FieldStorage set the default Content-Type for POST
+            # requests to a "traditional" value.
```

Comment:
I'm not sure I understand this comment.  Does this mean "cgi.FieldStorage used to set Content-Type if it was missing, so we do that here, now that we don't use cgi.FieldStorage any more"?

Reply:
Yes, exactly.  I've clarified that comment.